graphql/chunker.js

```diff
@@ -46,6 +59,24 @@ const getSemanticChunks = (text, chunkSize) => {
   const breakBySentences = (str) => breakByRegex(str, /(?<=[.。؟！?!\n])\s+/, true);
   const breakByWords = (str) => breakByRegex(str, /(\s,;:.+)/);
 
+  const breakByHtmlElements = (str) => {
+    const $ = cheerio.load(str, null, true);
+
+    // the .filter() call is important to get the text nodes
+    // https://stackoverflow.com/questions/54878673/cheerio-get-normal-text-nodes
+    let rootNodes = $('body').contents();
+
+    // create an array with the outerHTML of each node
+    const nodes = rootNodes.map((i, el) => $(el).prop('outerHTML') || $(el).text()).get();
+
+    // remove newlines from each node
+    nodes.forEach((node, i) => {
+        nodes[i] = node.replace(/\r?\n|\r/g, " ").trim();
```

Comment:
Why?  Is this losing formatting potentially?  Why not?

Reply:
In general, within an HTML tag, whitespace doesn't mean anything.

1. <p>Hi,

How are you?</p>

is the same as
2. <p>Hi, How are you?</p>

As a result, the LLM may respond with 2. if you send it 1. So removing newlines and trimming is a way to normalize the input. 

But... your question made think of the one case where whitespace does matter, which would be if it's contained within a `pre` tag. I'll have to do some research to see if there's any others.